src/pages/TodoPage/TodoPageContainer.tsx

```diff
@@ -19,6 +25,44 @@ export const TodoPageContainer: React.FC = () => {
 
   const handleVisibleModal = () => {
     setVisibleModal((prev) => !prev);
+    setModalType('create');
+  };
+
+  const handleVisibleModalWithEditFeature = (id: string) => {
+    setVisibleModal(true);
+    setCurrIDForEdit(id);
+    setModalType('edit');
+    const currTodo = todos.find((todo) => todo.id === id);
+    setInitialValueForModalWithEditFeature({
+      id: currTodo!.id,
+      title: currTodo!.title,
+      description: currTodo!.description,
+    });
+  };
+
+  const onCloseModalWithEditFeature = () => {
+    setVisibleModal(false);
+  };
```

Comment:
Useless method. Replace it with 'handleVisibleModal' and nothing will be changed.
Create a handleCloseModal method and use it for closing modal. 
Also, try to pass on a method to the modal window that will be triggered on 'submit' action. Long story short. Modal window should get three parameters: onClose, onSubmit, initialValues

Reply:
tried to resolve the issue